Add Issue templates and Gradle configuration files (#2)
ballerina-platform/static-code-analysis-tool: README.md

# Ballerina Static Code Analysis Tool

## Overview

> Static Code Analysis (SCA) is an approach that uses tools to examine code without execution. They are used for identifying potential issues like bugs, vulnerabilities, and style violations. SCA improves software quality by early issue detection, ensuring better maintainability, and providing enhanced security.

Reviewer: ```suggestion
Static Code Analysis (SCA) uses tools to examine code without executing the code. They are used for identifying potential issues like bugs, vulnerabilities, and style violations. SCA improves software quality by detecting issues early, ensuring better maintainability, and providing enhanced security.
```

> The Ballerina scan tool provides a set of command line tools to statically analyze Ballerina files and report analysis issues.

Reviewer: We can elaborate a bit more here saying this is a Ballerina tool. 

> provides a set of command line tools

isn't this rather "provides the command-line functionality to ..."

Also elaborate more on what exactly this repo contains. i.e., the fact that this consists of the core scan logic and extension points for analysis and reporting to platforms

> The Ballerina scan tool is a Ballerina tool that provides the command-line functionality to statically analyze Ballerina files and report analysis results. The scan tool consists of core scan logic and extension points for introducing additional analysis and reporting results to static code analysis platforms.

Reviewer: Introduce this as Ballerina supports SCA using the Ballerina scan tool ....

Then instead of the second sentence, on a new line, let's say

This repository consists of 
- the Ballerina scan tool implementation
- the core scan logic
- extension points for ....

# Ballerina Static Code Analysis Tool

## Overview
Static Code Analysis (SCA) is an approach that uses tools to examine code without execution. 
It's used for identifying potential issues like bugs, vulnerabilities and style violations.
SCA improves software quality by early issue detection, creating better maintainability, and
providing enhanced security. The Ballerina scan tool provides a set of command line tools to
statically analyze Ballerina files and report analysis issues.

## Prerequisites
1. openJDK 17 ([Adopt OpenJDK](https://adoptium.net/temurin/releases/?version=17) or any other OpenJDK distribution)

2. [Ballerina version: 2201.8.5](https://ballerina.io/downloads/)

## Building from the source
Execute the commands below to build from the source.

1. Export GitHub Personal access token with read package permissions as follows,
    ```bash
    export packageUser=<Username>
    export packagePAT=<Personal access token>

>     ```

Reviewer 2: Shall we have the steps for windows OS as well? It'll make things easier for the windows users.

Author: +1, Fixed in [cf88522](https://github.com/ballerina-platform/static-code-analysis-tool/pull/2/commits/cf88522631113897d629d7a51feece25f60b9303)

# Ballerina Static Code Analysis Tool

> ## Overview

Reviewer: ```suggestion
## Overview

```

Reviewer: Please fix everywhere.

Static Code Analysis (SCA) is an approach that uses tools to examine code without execution. 

> It's used for identifying potential issues like bugs, vulnerabilities and style violations.

Reviewer: ```suggestion
It is used for identifying potential issues like bugs, vulnerabilities, and style violations.
```

```suggestion
They are used for identifying potential issues like bugs, vulnerabilities, and style violations.
```

SCA improves software quality by early issue detection, creating better maintainability, and

> providing enhanced security. The Ballerina scan tool provides a set of command line tools to

Reviewer: The line breaks don't seem natural. Why do we need this length?

statically analyze Ballerina files and report analysis issues.

## Prerequisites

> 1. openJDK 17 ([Adopt OpenJDK](https://adoptium.net/temurin/releases/?version=17) or any other OpenJDK distribution)

Reviewer: ```suggestion
1. OpenJDK 17 ([Adopt OpenJDK](https://adoptium.net/temurin/releases/?version=17) or any other OpenJDK distribution)

```

> 2. [Ballerina version: 2201.8.5](https://ballerina.io/downloads/)

Reviewer: Shouldn't be tied to a b7a version since this is the repo for active development.

## Building from the source
Execute the commands below to build from the source.

1. Export GitHub Personal access token with read package permissions as follows,

   Linux/Unix
   ```bash
    export packageUser=<Username>
    export packagePAT=<Personal access token>
   ```

   Windows
   ```batch
   set packageUser=<Username>

>    set packagePAT=<Personal access token>

Reviewer: Why do we need these now?
